lib/utils.js

```diff
@@ -204,12 +204,18 @@ is_valid_year = function(year, minYear, maxYear) {
 Checks that the number representing the day of the month is within a valid range
 
 :param string day: day to be validated
+:param string month: Optional. Month to be validated.
+:param string year: Optional. Year to be validated.
 */
-is_valid_day_of_month = function(day) {
-    // check that it is a number and between 1 and 31
-    return this.check_valid_number(day)
-        && parseInt(day, 10) >= 1
-        && parseInt(day, 10) <= 31;
+is_valid_day_of_month = function(day, month, year) {
+    if (month && year) {
+        return this.is_valid_date(year + "-" + month + "-" + day, "YYYY-MM-DD");
+    } else {
+        // check that it is a number and between 1 and 31
+        return this.check_valid_number(day)
+            && parseInt(day, 10) >= 1
+            && parseInt(day, 10) <= 31;
+    }
```

Comment:
You could further improve this function by checking for validity if the day and month are provided, but not the year. So it would check for 30 / 31 days in the appropriate months, and <= 29 for February.